packages/sitecore-jss/src/layout/graphql-layout-service.ts

```diff
@@ -94,20 +83,11 @@ export class GraphQLLayoutService extends LayoutServiceBase {
    * @returns {GraphQLClient} implementation
    */
   protected getGraphQLClient(): GraphQLClient {
-    if (!this.serviceConfig.endpoint) {
-      if (!this.serviceConfig.clientFactory) {
-        throw new Error('You should provide either an endpoint and apiKey, or a clientFactory.');
-      }
-
-      return this.serviceConfig.clientFactory({
-        debugger: debug.layout,
-        retries: this.serviceConfig.retries,
-        retryStrategy: this.serviceConfig.retryStrategy,
-      });
+    if (!this.serviceConfig.clientFactory) {
+      throw new Error('clientFactory needs to be provided when initializing GraphQL client.');
```

Comment:
Can you, please, double check that this error message is the same across all the services? I've seen at least one different in _graphql-dictionary-service_